### ISSUE TITLE
Failures with #anchors: file not found

### ISSUE THREAD

**Issue by grahamc** [lycheeverse/lychee#1591]
I have a website with this in the HTML:

```html
<h2 id="outcomes" class="heading"> <a href="#outcomes"> Outcomes </a> </h2>
```

Using lychee main (1780aa0daa70462e296f9811e06178f13fd945db, v0.18.0) or lychee v0.17.0, and calling it like this:

```
lychee --threads 1 --cache --config ./lychee.prod.toml --verbose --base "$(realpath ./dist)" ./dist
```

and a lychee.prod.toml containing:

```toml
cache = true
include_mail = true
max_cache_age = "1w"
max_redirects = 5

exclude = [
# a few unrelated entries...
]
```

I get this error:

```
[ERROR] file:///Users/grahamc/src/github.com/DeterminateSystems/website/dist/%23outcomes | Cannot find file
```

**Comment by mre** [comment]
Can you try 0.16.0? Might be a duplicate of #1457, https://github.com/lycheeverse/lychee/issues/1577, https://github.com/lycheeverse/lychee/issues/1574. I basically broke anchors/fragments. 😬

**Comment by grahamc** [comment]
I'll give it a try, @mre! Thank you! By the way, while I did just open up three issues, we are very happy with Lychee. It works very well for us, and I'm grateful for the project. Thank you.

**Comment by grahamc** [comment]
You're right, totally solved on v0.16.0! Thank you!
